packages/xo-server-web-hooks/src/index.js

```diff
@@ -0,0 +1,113 @@
+import request from 'http-request-plus'
+import { groupBy, mapValues } from 'lodash'
+
+const makeRequest = (url, type, data) =>
+  request(url, {
+    body: JSON.stringify({ ...data, type }),
+    headers: { 'content-type': 'application/json' },
+    method: 'POST',
+    onRequest: req => {
+      req.setTimeout(1e4)
+      req.on('timeout', req.abort)
+    },
+  })
+
+class XoServerHooks {
+  constructor({ xo }) {
+    this._xo = xo
+
+    // Defined in configure().
+    this._conf = null
+    this._key = null
+    this.handlePreHook = (...args) => this.handleHook('pre', ...args)
+    this.handlePostHook = (...args) => this.handleHook('post', ...args)
```

Comment:
`handleHook` does not need to be defined as a class method.

```suggestion
    this.handlePostHook = handleHook.bind(this, 'post')
```